### PR TITLE
Doc + cleanup Hook class

### PR DESCRIPTION
**Pull request by firstred**
| Questions | Answers |
| --- | --- |
| Branch? | `develop` |
| Description? | Upgrading the PHPDoc comments of this class, using the latest PS standards and also applying the PS standards to some of the code. Also getting the Franglais out of the comments, getting rid of things the Symfony code sniffer nags about, etc. It might have some additional `todo` tags, depending on whether the code was comprehensible or not. |
| Type? | improvement |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | N/A |
| How to test? | N/A |

Warning: this PR also has a code cleanup. Please check this first before merging!

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
Would you mind to rebase it against develop branch ?

File: classes/Hook.php [hunk 1]
```diff
@@ -263,182 +289,212 @@ public static function getHookModuleList()
...
+
         return !empty($rows);
     }
 
-    public static function registerHook($module_instance, $hook_name, $shop_list = null)
+    /**
+     * Register Hook
+     *
+     * @param Module     $moduleInstance
+     * @param string     $hookName
```
Comment (thierrymarianne) [inline classes/Hook.php]: It appears $hookName could be passed as an array.

**Review by thierrymarianne — CHANGES_REQUESTED** [review]
Please see minor change requested.

**Comment by firstred** [comment]
Updated

**Pull request closed without merging**